pkg/sql/grant_revoke_system.go

```diff
@@ -426,20 +398,64 @@ func synthesizePrivilegeDescriptorFromSystemPrivilegesTable(
 	// grant. If there is an empty row for Public, then public
 	// does not have grant.
 	if spo.GetObjectType() == privilege.VirtualTable {
-		if _, found := privileges.FindUser(username.PublicRoleName()); !found {
-			privileges.Grant(username.PublicRoleName(), privilege.List{privilege.SELECT}, false)
+		if _, found := privDesc.FindUser(username.PublicRoleName()); !found {
+			privDesc.Grant(username.PublicRoleName(), privilege.List{privilege.SELECT}, false)
 		}
 	}
 
 	// We use InvalidID to skip checks on the root/admin roles having
 	// privileges.
-	if err := privileges.Validate(
+	if err := privDesc.Validate(
 		descpb.InvalidID,
 		spo.GetObjectType(),
 		spo.GetPath(),
 		privilege.GetValidPrivilegesForObject(spo.GetObjectType()),
 	); err != nil {
 		return nil, err
 	}
-	return privileges, err
+	return privDesc, err
+}
+
+func accumulatePrivilegeDescriptorFromSystemPrivilegesRow(
```

Comment:
how would you feel about making a little struct and making a method to add a new row to it? There's something a little offputting about the way we keep passing the same object here but then we also stick it into the map. If we had some sort of accumulator struct, it'd feel a little cleaner to me.

Reply:
that seems good